src/plugins/network/proxy/common.ts

```diff
@@ -0,0 +1,113 @@
+// File size is not recommended to exceed 6M,
+
+import {
+  isBlob,
+  isDocument,
+  isFile,
+  isFormData,
+  isURLSearchParams,
+} from 'src/utils';
+import { SpyNetwork } from 'types';
+
+// 10M files would result negative performance impact distinctly in local-test.
+export const MAX_SIZE = 1024 * 1024 * 6;
+export const Reason = {
+  EXCEED_SIZE: 'Exceed maximum limit',
+};
+
+export const BINARY_FILE_VARIANT = '(file)';
+export function formatEntries(data: IterableIterator<[string, unknown]>) {
+  const result: [string, string][] = [];
+  let processor = data.next();
+  while (!processor.done) {
+    const [key, value] = processor.value;
+    let variant: string;
+    if (isFile(value)) {
+      variant = BINARY_FILE_VARIANT;
+    } else {
+      variant = String(value);
+    }
+    result.push([key, variant]);
+    processor = data.next();
+  }
+  return result;
+}
+
+export function resolveUrlInfo(target: URL | string) {
+  try {
+    const { href, pathname, search, searchParams, hostname } = new URL(
+      target,
+      window.location.href,
+    );
+    const url = href;
+    const query = [...searchParams.entries()];
+    // https://exp.com => "exp.com"
+    // https://exp.com/ => "exp.com"
+    // https://exp.com/devtools => "devtools"
+    // https://exp.com/devtools/ => "devtools"
+    // https://exp.com/devtools?version=Mac/10.15.7 => "devtools?version=Mac/10.15.7"
+    // https://exp.com/devtools/?version=Mac/10.15.7 => "devtools?version=Mac/10.15.7"
+    let name = pathname.replace(/[/]*$/, '').split('/').pop() || '';
+    name += search;
+    if (name === '') {
+      name = hostname;
+    }
```

Comment:
I have tried using regexp to extract the last path, but for the sake of code readability, I finally keep the previous implementation, do you have greater idea? @YanagiEiichi